### ISSUE TITLE
About GlideN64´s future

### ISSUE THREAD

**Issue by AlexN64gamer** [gonetz/GLideN64#1848]
I think the future development should focus on getting the best possible emulation for games that still have issues like Mario Tennis, Indiana Jones, Battle for Naboo, etc, and I think there are a couple of other games, some japanese ones maybe that arent still well optimized, all in all, the plugin is, for me, the best that there is out there, cause Angrylions. although it is almost perfect, its still very slow in mid-systems, and some high end ones to, so it makes it practically not a good option, some of the issues ive come of playing Mario Tennis, are glitched text in screen when a player wins a match, koopa troppa, and bomb-on "ball boys" dissapearing when then N64 depth compare option is enabled, and shadows not being properly displayed when "original n64 resoultion" is not selected, awesome work, and great plugin guys ;) keep the good work and hope my suggestion will be take in count

**Comment by DonelBueno** [comment]
Would it be possible to improve the framerate over the original hardware? Goldeneye 007 is almost unplayable in its original framerate.

**Comment by AmbientMalice** [comment]
>Would it be possible to improve the framerate over the original hardware? Goldeneye 007 is almost unplayable in its original framerate.

That's not controlled by the graphics plugin. Running those games with a Counter Factor of 1 boosts the framerate significantly.

**Comment by GamingTVYT** [comment]
@AlexN64gamer Vigilante 8 for example ...

**Comment by DonelBueno** [comment]
@AmbientMalice wow, I realized there is an overclock option on PJ64 newer builds, it works really well. Do you know if it causes accuracy issues?

**Comment by LegendOfDragoon** [comment]
@DonelBueno , overclocking can cause games to run too fast. Like in smash bros, the menus load faster with OC enabled. You should use whatever settings you think are best. Imo "accuracy" is not the most important thing. Use whatever settings give you the most enjoyable experience.

**Comment by retrobenny** [comment]
There is a minor bug on Perfect Dark where one of the logo snaps around weirdly compared to having no such issue on 1964's overclock option.

**Comment by AlexN64gamer** [comment]
Its really cool to see your comments guys, I realize how many of us like emulation and features that could be implemented for next releases, I personaly will always thank all the people that developed GlideN64, because N64 is the console of my childhood and i LOVE it, to the point of always encouraging 100% accurate emulation of the hardware to make myself believe im playing the real thing xD (I cant save enough maney nowadays to buy an actual N64 and the games) I use an usb adapter for my actual N64 controller to have the most accurate experience with emulators, I wrote this cause I thought it would be interesting to make developers know we like what has been done so far, and that we support their work and´d like to see the plugin improve more and more, there are still lot of graphic issues, in many games, some of them that are console classics (Like Mario Tennis or Vigilante 8(@Gaming TVYT)) but im sure with the time it´ll become the best in its kind, as I said before, Angrylions is not for general use yet, it is almost 100 percent accurate, but it doesnt run even close to 100 percent fps (at least in my pc) thanks for your comments

**Comment by LuismaSP89** [comment]
I totally agree with GlideN64 optimization, I use Jabo plugin for play Goldeneye, since with GlideN64 is totally unplayable due to the low fps. Other games runs great, but there´s some speed issues (For example in mario tennis, when a match starts, the lightning fall between the two players is a framerate killer even in my high-end PC.

**Comment by theboy181** [comment]
I don’t recall there being any issues with performance when using GLideN64 and those games. 

What settings are you using in your emulators and in GLideN64?

**Comment by LegendOfDragoon** [comment]
>I don’t recall there being any issues with performance when using GLideN64 and those games.

Of course a gtx 1080 won't struggle with N64 emulation :smile: .

@LuismaSP89 how's the performance with FBE disabled? I like to disable it now, after realizing how much better the performance is. I don't really need it in many of the games i play.

**Comment by theboy181** [comment]
I didn’t have any issues with my old laptop either. Why do t you test it out?

**Comment by DonelBueno** [comment]
@LuismaSP89 You probably have xBRZ texture scaling activated, try disabling it.

**Comment by LuismaSP89** [comment]
I got a GTX 970, 16GB of RAM, and i7 4790k 4,4Ghz, but for me those games (Goldeneye and the lightning in Mario tennis) have a big slowdown all the time. Other games have a small-medium slowdown with max settings, but this two are special.

@LegendOfDragoon In the case of Goldeneye and Mario tennis, if you disable framebuffer emulation, the games have totally broken graphics. In any case, I tried disabling it and there´s no improvement in performance, at least for me.

After many testing, there are 3 guilty (In order of resource eating from less to more):

-Disabling the checkbox "Enable N64-style mip-mapping (With this disabled, games look awesome at the far distance)

-Enable some filter under (Texture enhancement) "Sharp filtering or smooth"

-Enable the "Enhancement" filters, specially the 6xbrz is the ultimate monster of resource-eating, but even with 2xbrz and the other two options above, my high-end PC shits in his pants.

I know that specially the filters require a great PC, but maybe these three options can be improved to run faster (I know nothing from programming, to be honest)

**Comment by StenApp** [comment]
@olivieryuyu
I'm curious, can you please tell which problems are still pending?

> After Indy, there is still many things in HLE to do:
> Batte of Naboo specific commands
> HVQ microcode (if really need)
> Fog issue in Zelda OOT
> S2DEX issues in various games
> Dark Rift issues
> 

-I think _Batte of Naboo specific commands_ is done
-S2DEX  might be solved here #133 

What about the rest?

**Comment by olivieryuyu** [comment]
HVQ: Gillou started to work on it
S2DEX: #133 and #936 is what I am working on currently.

The rest is still to be started.

All issues are very hard to understand so it takes time.

**Comment by StenApp** [comment]
Thanks for the quick answer. I wonder why nobody solved this before...

**Comment by StenApp** [comment]
@Gillou68310 Hey, just wanted to know what the status of the HVQ microcode is.

**Comment by StenApp** [comment]
@gonetz 
what's next, now that most of the bugs are fixed?

**Comment by oddMLan** [comment]
I guess this can be closed.

**Comment by gonetz** [comment]
Well, it is not an issue, rather some kind of chat. It does not bother me.

**Comment by weinerschnitzel** [comment]
To make this issue more useful as a chat, I am curious where current priorities are. A lot of progress has been made.

I would like to see the following addressed and I think coverage emulation is getting close to consideration.
1. Regressions
2. Texture coordinate refactoring, 2d padding/scaling/filtering issues
3. HLE fixes for RE2 and HVQM support (or collaboration to support)
4. Hardware LLE completion
5. Angrylion LLE compute shader implementation?
6. Coverage emulation?

I'm not sure if 5 would be necessary if 4 is perfected, or if 5 can solve issues 4 cannot. High resolutions is curious if 5 can handle.

As mentioned, I think the project is getting to a point where coverage can be road mapped.

What are everyone else's thoughts?

**Comment by loganmc10** [comment]
Why not just be ok with gonetz working on whatever he wants to work on? It's a hobby project mostly done by 1 person you guys, gotta manage those expectations

**Comment by LuismaSP89** [comment]
@weinerschnitzel Points 1 and 4 seems interesting. But the most-wanted for me in the future's of the plugin. Is to solve Goldeneye's problems, starting with the micro-stutters when you switch between weapons, and other issues like a general slowness in the game when are played with 3x IR or some enhancements even using an strong PC.

**Comment by weinerschnitzel** [comment]
@loganmc10 No expectations here. Just curious what people want to see and what bigger ticket items people think are approachable next.

**Comment by gonetz** [comment]
`Regressions`
As usual, it is top priority task.

`Texture coordinate refactoring, 2d padding/scaling/filtering issues`
@standard-two-simplex made huge step in that direction: #2202
I call everyone to support him with throughout testing.

`HLE fixes for RE2 and HVQM support (or collaboration to support)`
HLE fixes for RE2 are mostly done, see #2152
Regarding HVQM support, I'm not sure what is expected from me.

`Hardware LLE completion`
I sick of LLE.

`Angrylion LLE compute shader implementation?`
Why?
* Angrylion LLE compute shaders will work only in LLE.
* They will not solve issues with frame buffer effects.

`Coverage emulation?`
I'm constantly thinking about it.
Alas, there are more important tasks.

**Comment by olivieryuyu** [comment]
From my point of view, we do not need perfect LLE as GlideN64 was developed as a HLE plugin and GE007/PD are fixed with the ongoing and much better implementation that the ziggy one :)

Just my opinion of course :)

**Comment by retrobenny** [comment]
The main goal for GLideN64 was to have the most accurate HLE graphics plugin possible and its done excellent strides since its inception and has been awesome for enhancing graphics quality as well on top of having high performance for something meant to be accurate.

**Comment by StenApp** [comment]
@gonetz do you have any plans on when to release GlideN64 5.0?
Any milestones to reach?

**Comment by gonetz** [comment]
Good point. I need to update my Patreon page at least.
The main requirement for a new public release is no regressions.
Currently there are few open regressions.
Then I plan to finally implement the customizable hot-keys feature.
And we still have ~250 issues to fix.
